pygmt/base_plotting.py

```diff
@@ -994,3 +994,353 @@ def text(
+                    "p_plunge",
+                    "exponent",
+                ],
+            }
+
+            # to keep track of where optional parameters exist
+            data_pointers = {
+                "lon": lon,
+                "lat": lat,
+                "depth": depth,
+                "plot_lon": plot_lon,
+                "plot_lat": plot_lat,
+            }
+
+            # make a DataFrame copy to check convention if it contains
+            # other parameters
+            if isinstance(spec, (dict, pd.DataFrame)):
+                # check if a copy is necessary
+                copy = False
+                drop_list = []
+                for pointer in data_pointers:
+                    if pointer in spec:
+                        copy = True
+                        drop_list.append(pointer)
+                if copy:
+                    spec_conv = spec.copy()
+                    # delete optional parameters from copy for convention check
+                    for item in drop_list:
+                        del spec_conv[item]
+                else:
+                    spec_conv = spec
+
+            # set convention and focal parameters based on spec convention
+            convention_assigned = False
+            for conv in PARAM_CONVENTIONS:
+                if set(spec_conv.keys()) == set(PARAM_CONVENTIONS[conv]):
+                    convention = conv.lower()
+                    foc_params = PARAM_CONVENTIONS[conv]
+                    convention_assigned = True
+                    break
+            if not convention_assigned:
+                raise GMTError(
+                    "Parameters in spec dictionary do not match known " "conventions."
+                )
+
+            # create a dict type pointer for easier to read code
+            if isinstance(spec, dict):
+                dict_type_pointer = list(spec.values())[0]
+            elif isinstance(spec, pd.DataFrame):
+                # use df.values as pointer for DataFrame behavior
+                dict_type_pointer = spec.values
+
+            # assemble the 1D array for the case of floats and ints as values
+            if isinstance(dict_type_pointer, (int, float)):
+                # update pointers
+                set_pointer(data_pointers, spec)
+                # look for optional parameters in the right place
+                lon, lat, depth, plot_lon, plot_lat = update_pointers(data_pointers)
+
+                # Construct the array (order matters)
+                spec = [lon, lat, depth] + [spec[key] for key in foc_params]
+
+                # Add in plotting options, if given, otherwise add 0s as
+                # required by GMT
+                for arg in plot_lon, plot_lat:
+                    if arg is None:
+                        spec.append(0)
+                    else:
+                        if "C" not in kwargs:
+                            kwargs["C"] = True
+                        spec.append(arg)
```

Comment:
I didn't read the codes carefully. Just want to let you know that, since GMT 6.1, plot_lon and plot_lat are not required.

Reply:
Got it